src/sys.rs

```diff
@@ -270,6 +271,81 @@ mod inner {
     impl Eq for SteadyTime {}
 }
 
+#[cfg(target_env = "sgx")]
+mod inner {
+    use std::ops::{Add, Sub};
+    use Tm;
+    use Duration;
+    use super::common::{time_to_tm, tm_to_time};
+    use std::time::SystemTime;
+
+    /// The number of nanoseconds in seconds.
+    const NANOS_PER_SEC: u64 = 1_000_000_000;
+
+    #[derive(Copy, Clone, Debug, PartialOrd, Ord, PartialEq, Eq)]
+    pub struct SteadyTime {
+        t: Duration
+    }
+
+    pub fn time_to_utc_tm(sec: i64, tm: &mut Tm) {
+        time_to_tm(sec, tm);
+    }
+
+    pub fn time_to_local_tm(sec: i64, tm: &mut Tm) {
+        // FIXME: Add timezone logic
+        time_to_tm(sec, tm);
+    }
+
+    pub fn utc_tm_to_time(tm: &Tm) -> i64 {
+        tm_to_time(tm)
+    }
+
+    pub fn local_tm_to_time(tm: &Tm) -> i64 {
+        // FIXME: Add timezone logic
+        tm_to_time(tm)
+    }
+
+    pub fn get_time() -> (i64, i32) {
+        SteadyTime::now().t.raw()
+    }
+
+    pub fn get_precise_ns() -> u64 {
+        let duration = SteadyTime::now().t.raw();
+        duration.0 as u64 * NANOS_PER_SEC + duration.1 as u64
```

Comment:
This is fine. My original suggestion was to do this one in terms of `SystemTime`, not `SteadyTime`, that would save you a signed-unsigned conversion.